app/controllers/application_controller.rb

```diff
@@ -1,3 +1,22 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+  
+  helper_method :current_user, :signed_in?
+  
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+  
+  def signed_in?
+    !!current_user
+  end
+  
+  def require_user
+    acess_denied unless signed_in?
+  end
+  
+  def access_denied
+    flash['error'] = "You can't do that."
```

Comment:
This app uses bootstrap 3 which uses "danger" instead of "error". "notice" is now "success" as well. You can see the other class names in the docs.